compiler/damlc/daml-stdlib-src/DA/Internal/Desugar.daml

```diff
@@ -35,4 +35,19 @@ data NonConsuming t = NonConsuming {}
 data PreConsuming t = PreConsuming {}
 data Consuming t = Consuming {}
 data PostConsuming t = PostConsuming {}
-data Implements t i = Implements
+
+class Implements t i where
+  toInterface : t -> i
+  toInterface = GHC.Types.primitive @"EToInterface"
```

Comment:
Note that I simply added the implementations here since they're always the same set of primitives. This way, the generator only needs to make empty instances. However, please let me know if there's a good reason not to do it this way

Reply:
ah, it seems that some of the primitives only work with concrete types, so I'll go back to the class methods being defined by the instances.

https://github.com/digital-asset/daml/blob/21c97453bd5370bfaeec69afece2ec56659ef304/compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/Primitives.hs#L458-L487

Reply:
yeah one of the reasons for going via the class dance is to be able to use the concrete types in the instances.